### DISCUSSION TITLE
Contents of configuration file

### DISCUSSION THREAD

**Discussion by kjogr12** [RMeli/gnina-torch#42] [Q&A] [answered]
Hi.

I would like to use this repository and I have a question.
When training, what should I write as settings in the types file?
ex. https://github.com/RMeli/gnina-torch/blob/main/tests/data/test.types

I reckon the first column is label information, the fourth and fifth columns are PDB and ligand information, but what are the second and third columns?

**Comment by RMeli** [comment, marked as answer]
Hello.
Sorry for the scarce documentation, but the repository is still very much WIP. The `.types` file you linked is a test file, therefore it contains all possible labels.

Currently `gnina-torch` supports the following combinations:
* Ligand pose prediction: you only need the pose label (`--label_pos `)
* Ligand pose and binding affinity prediction (multi-task): you need the pose label (`--label_pos`) and the affinity label (`--affinity_pos`)
* Ligand pose and flexible residues pose prediction (multi-task): you need the ligand pose label (`--label_pos`) and the flexible residues pose label (`--flexlabel_pos`)

The CLI arguments `--label_pos` (default `0`), `--affinity_pos` (default `1`), and `--flexlabel_pos` (experimental!) allow you to tell `gnina-torch` in which column of the `.types` file the different quantities are located.

The last point in particular is still very experimental and under development/study. 

If you only want to train for pose prediction, the following format is sufficient for the `.types` file:
```text
pose_label protein_file ligand_file
```
If you want to train for both pose and binding affinity prediction, you need to use the following format for the `.types` file:
```text
pose_label binding_affinity protein_file ligand_file
```

**Reply by kjogr12** [reply]
Thank you for the detailed explanation.
I understand the arguments and how to specify them. Although it is WIP, I can do what I want to do now with the current features, so I will give it a try!
